### ISSUE TITLE
No members in group idm_all_persons

### ISSUE THREAD

**Issue by joostrijneveld** [kanidm/kanidm#2873]
### I did this

I created a user account. Calling `kanidm person get` shows that they're a `memberof: idm_all_persons`.

### I expected the following

Calling `kanidm group list-members idm_all_persons` to list them.

### This happened instead

`2024-07-05T09:35:02.544568Z  WARN kanidm_cli::group: No members in group idm_all_persons`

### Kanidm version details

- Output of `kanidm(d) version`: 1.2.3
- Are you running it in a container? If so, which image/tag?: Yes, `:latest`
- If not a container, how'd you install it:
- Operating System / Version (On Unix please post the output of `uname -a`):

Ubuntu LTS 22.04 on the server, macOS 14.5 on the client

**Comment by Firstyear** [comment]
How did you make the user and what accounts were you using in the process to view the user?

**Comment by Firstyear** [comment]
```
login -D idm_admin
person create steve Steve
person get steve
> memberof: idm_all_persons@localhost
> memberof: idm_all_accounts@localhost
> memberof: idm_people_self_name_write@localhost

group get idm_all_persons
> dynmember: steve@localhost
```

I'm not seeing the same behaviour (at least, not on git master)

**Comment by joostrijneveld** [comment]
I've created a person using `person create` using `idm_admin`, added them to `idm_people_admins` and subsequently used that person to create a second person using `person create`. Neither show up in `group list-members`.

I can reproduce what you're doing, and indeed in the output of `group get idm_all_persons` they show up just fine. However, when I call `group list-members`, I get the above error. Is this perhaps a semantic misunderstanding, i.e., are 'members' in 'list-members' only subgroups and not persons?

**Comment by yaleman** [comment]
`list-members` does a direct pull for the attribute `member` from the group which ... probably doesn't calculate dynmembers? 🤔

**Comment by joostrijneveld** [comment]
I’ll immediately admit that I don’t know the difference between a member and a dynmember.

**Comment by yaleman** [comment]
`dynmembers` are dynamic members of the groups, instead of being direct members they're part of other groups etc. It was more a comment for @Firstyear 😄

**Comment by Firstyear** [comment]
> I’ll immediately admit that I don’t know the difference between a member and a dynmember.

The intent is that a dynmember is calculated and added to the group, but an admin can *manually* adjust the member attribute to extend the group further so that it's both the output of the query/filter and manually added members. By separating these, we know who has authority over the attributes presence on the group.

**Comment by Firstyear** [comment]
Depends on #211

**Comment by Firstyear** [comment]
Well, depends on #211 unless I add a custom member get handle which I want to avoid more "hacks" like that. But I don't think this is hugely urgent or pressing to resolve? Do you have a use case you need satisfied?

**Comment by joostrijneveld** [comment]
Thanks for your comments!

No, this is not at all pressing to resolve; I came across this as I was debugging a misconfiguration and was simply pacing through `—help` looking for ways to introspect the state of the group I was relying on. As you described, `group get` also shows the (dyn)members.

**Comment by Firstyear** [comment]
Yep, as you noted dynmembers and members are both reflected in the inverse "memberof" attribute of all entries, and honestly, it's the best place to check for membership - since you already have the "account" the memberof is right there after all :)
